### PR TITLE
Optimization

### PR DESCRIPTION
**Pull request by cfrontin**
Delete the text explanations below these headers and replace them with information about your PR.
Please first consult the [developer guide](https://weis.readthedocs.io/en/latest/how_to_contribute_code.html) to make sure your PR follows all code, testing, and documentation conventions.

## Purpose

Additions to the optimization branch that are coming in from my fork.

These include:

- improved optimizer documentation
- improved optimization monitoring/evaluation capabilities
- more exploratory example 17
- vizualization toolbox created

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
Explain the steps needed to test the new code to verify that it does indeed address the issue and produce the expected behavior.

## Checklist
_Put an `x` in the boxes that apply._

- [ ] I have run existing tests which pass locally with my changes
- [ ] I have added new tests or examples that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

File: examples/17_IEA22_Optimization/analysis.ipynb
Comment (dzalkind) [inline examples/17_IEA22_Optimization/analysis.ipynb:1]: Do you want to keep this notebook in the repo?  What is it analyzing?

File: examples/17_IEA22_Optimization/run.sh
Comment (dzalkind) [inline examples/17_IEA22_Optimization/run.sh:1]: Same for this bash script?  I'm curious...what is this doing?

Comment (cfrontin) [reply examples/17_IEA22_Optimization/run.sh:1]: this is just my run script that manages different optimizers. I've taken it out as of f9d4cef6441395c3103cb600d1618d39650a6703, and added README.md such that users can recreate its functionality on an _a la carte_ basis.

File: examples/17_IEA22_Optimization/analysis.ipynb
Comment (cfrontin) [reply examples/17_IEA22_Optimization/analysis.ipynb:1]: this notebook demonstrates the convergence behavior of COBYLA, SLSQP, and DE using the viz toolbox. I briefly touched base with you on its inclusion and will do so further irl.

**Pull request merged**